lib/renogen/change_log/validator.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Renogen
+  module ChangeLog
+    # Validates the change log
+    class Validator
+      def initialize(formatter)
+        @formatter = formatter
+        @validations = formatter.options['validations']
+      end
+
+      # Validates the change log
+      #
+      # @param changelog [ChangeLog::Model]
+      def validate!(changelog)
+        validate_headings(changelog)
+      end
+
+      protected
+
+      attr_reader :formatter, :validations
+
+      def validate_headings(changelog)
+        return if validations.nil?
+        return if changelog.items.none? { |item| validations.key?(item.group_name) }
+
+        validate_properties(changelog)
+      end
+
+      private
+
+      def validate_properties(changelog)
+        invalid_items = []
+        validations.each do |heading, values|
```

Comment:
Did you consider using a Regular expression.  I think this would provide more flexibility